### DISCUSSION TITLE
Running job_queue callback function can be stopped by CallbackQueryHandler?

### DISCUSSION THREAD

**Discussion by tualli-shlee** [python-telegram-bot/python-telegram-bot#3402] [Q&A]
Hi all, I wanna ask a question about` job_queue` and `CallbackQueryHandler`.

I'm working with the latest version of PTB (v20.0.a6).

I'm using `job_queue.run_repeating` to send messages repeatedly.
I want to make the `InlineKeyboardButton` callback function stop `run_repeating`.
I want to stop sending the message if I click the button during that time.
I tried to make CallbackQueryHandler run the function that removes every job in the job_queue, but it couldn't stop the currently running job.

If needed, I will upload some codes I tried.

Thanks

**Comment by Bibo-Joshi** [comment]
I'm not sure if that's easily possible. The `JobQueue` is build with the `APScheduler` library and AFAIK that library doesn't provide you with a way to get the jobs that are currently being executed.

What you could do is store a flag (or even `asyncio.Event`) somewhere (maybe in `bot_data`) and set that once the button is pressed. Within your job callback, check that flag before sending messages and `return` if it is set

**Comment by Poolitzer** [comment]
Hey, the usual way I do that is to give the job a specific name in `run_repeating` and then I can use `context.job_queue.get_jobs_by_name` with that name and on that return I can use `job.schedule_removal`. Mind you that it returns a list of jobs so iterate over it or use `[0]` if its only ever one job.
